packages/gatsby/src/cache-dir/component-renderer.js

```diff
@@ -67,11 +77,16 @@ class ComponentRenderer extends React.Component {
 
   render() {
     if (this.state.pageResources) {
-      return createElement(this.state.pageResources.component, {
-        key: this.props.location.pathname,
-        ...this.props,
-        ...this.state.pageResources.json,
-      })
+      return createElement(
```

Comment:
so layouts can now change on route changes, so they need to be handled within the pageResources state, I we could create another stateful component but IMO this cleaner.